Sources/Common/Core/LookupTable/index.js

```diff
@@ -61,7 +61,8 @@ function vtkLookupTable(publicAPI, model) {
   publicAPI.usingLogScale = () => false;
 
   //----------------------------------------------------------------------------
-  publicAPI.getNumberOfAvailableColors = () => model.table.length / 4;
+  // Remove special colors (min, max, NaN) from available colors
```

Comment:
Number does not include special colors (`min`, `max` and `NaN`).

Reply:
The table length does include these special colors, because in `forceBuild`, the length of the table is defined as such:
```js
const maxIndex = model.numberOfColors - 1;
// ...
model.table.length = 4 * maxIndex + 16;
```
So this function simply returns the `model.numberOfColors`. I could eventually simplify it by doing:
```js
publicAPI.getNumberOfAvailableColors =  publicAPI.getNumberOfColors;
```

And the special colors should not be included in `getNumberOfAvailableColors` because we add them by hand in `mapScalarsToTexture`. Also in vtkColorTransferFunction, it basically returns `model.nodes.length` which does not include NaN colors.

Reply:
All I meant was to rephrase the "comment" because mentionning "Remove" seems to say that there is a "side-effect" of calling this get function.